puma/hlplots/tagger.py

```diff
@@ -0,0 +1,214 @@
+"""Tagger module for high level API."""
+import h5py
+import numpy as np
+import pandas as pd
+from numpy.lib.recfunctions import structured_to_unstructured
+
+from puma.utils import calc_disc, logger
+
+
+class TaggerBase:  # pylint: disable=too-many-instance-attributes
+    """Base class handling tagger results."""
+
+    def __init__(self, model_name: str) -> None:
+        """Init Tagger class.
+
+        Parameters
+        ----------
+        model_name : str
+            Name of the model, also correspondinng to the pre-fix of the tagger
+            variables.
+        """
+        self.model_name = model_name
+        self.label = None
+        self.reference = False
+
+        self.scores = None
+        self.perf_var = None
+        self.flvs = ["pu", "pc", "pb"]
+
+        self.is_b = None
+        self.is_light = None
+        self.is_c = None
+
+    def extract_tagger_scores(
+        self, source: object, source_type: str = "data_frame", key: str = None
+    ):
+        """Extract tagger scores from data frame or file.
```

Comment:
I also tend to agree with @jobirk that this function should dynamically set `self.is_light` etc based on the passed/loaded jets

Reply:
this will create a big mess though, if it should be extended to more classes, I saw it more like puma should be more configured by the user while umami would handle these kind of things automatically.

It would require supporting cuts on 2 different label variables e.g. for leptonically decaying b-jets

Reply:
ok, is there a solution in the meantime for plotting from multiple different files containing potentially different jets?

The way I did this previously was to define a signal label signal class (e.g. `self.signal_class = 5`) and one or more background labels (e.g. `self.background_classes = [0, 4]`). Then you can avoid the need for separate `self.is_light` etc and just dynamically compute from the other members. Those could even be properties of the `Results` object and potentially using the class names instead of the label value (assuming there is somewhere a mapping between the two).

This would be a way to support b/c tagging with minimal code overhead

Reply:
@manuguth @jobirk, should we merge this as is and open an issue to discuss the multiple file support?